### DISCUSSION TITLE
[Q] Roadrunner slower than Nginx in production (Linux VM IP reverse proxied in Nginx)

### DISCUSSION THREAD

**Discussion by ndethi** [roadrunner-server/roadrunner#869] [Q&A]
When i run a benchmark locally in the VM, Roadrunner is miles ahead of Nginx in reqs/sec etc, but when I benchmark the public IP of the same app in production, it is far less impressive, in fact the same app hosted elsewhere on Nginx is faster. What factors could possibly be affecting this? Network latency, CPU?

**Comment by rustatian** [comment]
Hey @ndethi. It depends on many factors. Bare metal vs VM, Network latency, VM configuration, FS types, and many others.

**Reply by ndethi** [reply]
Ok got it. Comparing the hardware on both VMs I see that the box hosting the Nginx setup runs an AMD Epyc 32-Core Processor, 16GB Ram while RR is running on a Xeon E5 with 8 CPU Cores on 8 GB of ram. I take it then the Nginx setup with more cores will do better for concurrent requests? Would that be a factor?

**Reply by rustatian** [reply]
This is a completely wrong benchmark. If you wanted to get a fair result, create a VM, place the NGINX and RR on it. Turn off Nginx and turn on RR and vice versa. Then bench both. (simplified a lot)
We have a lot of public tests and posts where different people points, that RR is way faster than nginx or php-fpm (this is one of them: https://github.com/spiral/roadrunner/discussions/799#discussioncomment-1799003).
Creating a fair benchmark is a hard task with a lot of factors included. RR has workers which persist in the memory, while nginx/php-fpm has a dying model (I mean, a process created on every request).
